paasta_tools/kubernetes_tools.py

```diff
@@ -3432,3 +3434,44 @@ def update_crds(
             success = False
 
     return success
+
+
+def get_kubernetes_secret_name(service_name, secret_name, namespace="paasta"):
+    service = sanitise_kubernetes_name(service_name)
+    sanitised_secret = sanitise_kubernetes_name(secret_name)
+    name = f"{namespace}-secret-{service}-{sanitised_secret}"
+    return name
+
+
+def get_kubernetes_secret(secret_name, service_name, cluster):
+    k8s_secret_name = get_kubernetes_secret_name(service_name, secret_name)
+    template_name = f"{{.data.{secret_name}}}"
+
+    env = os.environ.copy()
+    env["KUBECONFIG"] = DEFAULT_KUBECTL_CONFIG_PATH
+
+    proc = subprocess.run(
+        [
+            "kubectl",
+            "get",
+            "secrets",
+            f"--cluster={cluster}",
+            f"-n=paasta",
+            f"{k8s_secret_name}",
+            f"--template={{{template_name}}}",
```

Comment:
Here we pass user input into a OS command, hence we are prone to command injection.  I would recommend to sanitize `cluster`, `k8s_secret_name` and `template_name` to avoid this vulnerability.
(I had a look at `sanitize_kubernetes_name` and what it does it doesn't help for this issue in particular.)

Reply:
There are multiple sources explaining how to prevent command injection. Of all, my favorite is [PortSwigger](https://portswigger.net/web-security/os-command-injection) (which I find better than OWASP here and there).

Reply:
It might not worth sanitizing it (other than best practice) if the following are true:
- this command would run on devbox
- the user already has access to the devbox
- the user has the same OS permissions as this service, so they wouldn't gain anything, other than potentially some repudiation

If the service has different permissions than devs (as in extra permissions), then it might worth implementing it.

I leave it up to you :)

Reply:
Added some sanitisation on alphanum chars and -, _ on all input.

Reply:
This is an exec call with no shell involved though, so you can't really do command injection.